apis/externalsecrets/v1beta1/secretstore_ibm_types.go

```diff
@@ -32,8 +32,23 @@ type IBMAuth struct {
 	SecretRef IBMAuthSecretRef `json:"secretRef"`
 }
 
+// +kubebuilder:validation:MinProperties=1
+// +kubebuilder:validation:MaxProperties=1
 type IBMAuthSecretRef struct {
 	// The SecretAccessKey is used for authentication
 	// +optional
 	SecretAPIKey esmeta.SecretKeySelector `json:"secretApiKeySecretRef,omitempty"`
+
+	SecretContainerAuth IBMAuthSecretContainerAuthSecretRef `json:"secretcontainerAuthSecretRef,omitempty"`
```

Comment:
I don't think this is a SecretRef (in the sense that it references a Secret). IMO this could be just `containerAuth`.

With the discussion above we probably should change the `auth` struct make sure that we are able to accommodate the different authentication mechanisms.

just an suggestion:
```yaml
apiVersion: external-secrets.io/v1beta1
kind: SecretStore
metadata:
  name: secretstore-sample
spec:
  provider:
    ibm:
      serviceUrl: "..."
      auth:
        containerAuth:
          # point explicitly to a file
          file:
            profile: "..."
            tokenLocation: "..."
            iamEndpoint: "..."
          # for the future
          serviceAccount:
            name: "..."
            namespace: "..."

        # already implemented stuff
        secretRef:
          secretApiKeySecretRef:
            name: ibm-secret
            key: apiKey
```

Reply:
This looks like a reasonable change @jack-evans and @knelasevero collaborated on the initial structure spec so I'd like at least one of them to agree this seems like a good change.

Regarding using the serviceAccount, I do not see any APIs exposed today that would allow this type of authentication in the IBM provider (today the closest thing would be to create a service account and provide its apikey in the existing secretRef flow), but since others have it, it does seem likely something similar will be added in the future, which this updated structure would support.

Reply:
I think when I originally looked at this I was going with consistency with the existing instantiation I.e this is config that is used for the connection to IBM Secrets Manager so it should be along the same lines as the existing configuration mechanism for Secrets Manager.

However, I agree that by having SecretRef you are implying there is a Kubernetes secret that has this configuration and the configuration is pointing to said secret. So I think that having just containerAuth is the right way to go... we just need to make sure it's well doc'd 🙂

Reply:
> Regarding using the serviceAccount, I do not see any APIs exposed today that would allow this type of authentication in the IBM provider for container or VPC app auth

i think i misunderstood how `containerAuth` works: I thought it would use a mounted `ServiceAccount` token, but it seems that IBM injects _some_ custom token via webhook. I'm not sure how that is implemented (happy if someone can point me to it if it is open source), and if we can build something similar like IRSA/WorkloadIdentity.
With that being said i think it doesn't make sense to plan for future but instead implement with what is documented (as described in the [comment above](https://github.com/external-secrets/external-secrets/pull/1177#discussion_r889812245))

```yaml
        containerAuth:
          # point explicitly to a file
          profile: "..."
          tokenLocation: "..."
          iamEndpoint: "..."
```